File: Diya/1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        StringBuilder str=new StringBuilder("");
        ListNode temp=head;
        while(temp!=null){
            str.append(temp.val);
            temp=temp.next;
        }
        return Integer.parseInt(str.toString(), 2);

    }
}

File: Karan/Linked list/Convert Binary Number in a Linked List to Integer.py
class ListNode:
 def __init__(self, val=0, next=None):
  self.val = val
  self.next = next
class Solution:
 def getDecimalValue(self, head):
  result = 0
  while head:
   result = (result << 1) | head.val
   head = head.next
  return result
head = ListNode(1, ListNode(0, ListNode(1)))
sol = Solution()
print(sol.getDecimalValue(head))